contracts/modules/capital/Pool.sol

```diff
@@ -389,16 +420,32 @@ contract Pool is IPool, MasterAwareV2, ReentrancyGuard {
     require(ok, "Pool: Transfer failed");
 
     // Transfer cover assets. Start from 1 (0 is ETH)
-    uint coverAssetsCount = coverAssets.length;
-    for (uint i = 1; i < coverAssetsCount; i++) {
+    uint coverAssetsCount = coverAssets.length - 1;  // substract 1 as ETH is skiped
+    uint investmentAssetsCount = investmentAssets.length;
+    uint totalAssetCount = coverAssetsCount + investmentAssetsCount;
+
+    SwapDetails[] memory _swapDetails = new SwapDetails[](totalAssetCount);
+    Asset[] memory _assets = new Asset[](totalAssetCount);
+    bool[] memory isCoverAsset = new bool[](totalAssetCount);
+
+    // Skip ETH
+    for (uint i = 1; i < coverAssetsCount + 1; i++) {
       _transferEntireAssetBalance(coverAssets[i].assetAddress, newPoolAddress);
+      _swapDetails[i - 1] = swapDetails[coverAssets[i].assetAddress];
+      _assets[i - 1] = coverAssets[i];
+      isCoverAsset[i - 1] = true;
     }
 
     // Transfer investment assets.
-    uint investmentAssetsCount = investmentAssets.length;
     for (uint i = 0; i < investmentAssetsCount; i++) {
       _transferEntireAssetBalance(investmentAssets[i].assetAddress, newPoolAddress);
+      _swapDetails[i + coverAssetsCount] = swapDetails[investmentAssets[i].assetAddress];
+      _assets[i + coverAssetsCount] = investmentAssets[i];
+      isCoverAsset[i + coverAssetsCount] = false;
     }
+
+    IPool(newPoolAddress).receiveAssets(_assets, _swapDetails, isCoverAsset);
```

Comment:
is it possible to just pass in the storage variables here or you need to construct all that stuff in memory first?